ProcessingShowPath/PathAnalyser.pde

```diff
@@ -0,0 +1,178 @@
+  private float wanderTimeSum;                             //sum of times when the player is away from the aid
+  
+  private int mistakes;  //amount of mistakes made when going back
+  private int shortcuts; //amount of shortcuts taken when going back
+  
+  Path path;
+  
+  PathAnalyser(Path path)
+  {
+    this.path = path;
+  }
+  
+  // analyse a single path
+  void analysePath()
+  {
+    findTimes();
+    FloatList wanderTimes = calculateTimesAwayFromBee(returnToAidTimes, wanderToAidTimes);
+    wanderTimeSum = floatListSum(wanderTimes);
+    noMovementTimeForward = floatListSum(forwardBreakTimes);
+    noMovementTimeBack = floatListSum(backBreakTimes);
+  }
+  
+  // print the result of the analysis
+  void printResult(boolean printBreakTimes)
+  {
+    String finishedString;
+    if (wanderToAidTimes.size() != 0 && wanderToAidTimes.get(0) == invalidTimeBecauseAidWasAbandoned)
+    {
+        finishedString = "Left aid";
+    }
+    else if(isBackTimeSaved)
+    {
+      finishedString = "Yes";
+    }
+    else 
+    {
+      finishedString = "No";
+    }
+    forwardTime -= noMovementTimeForward;  //Adjust for participants taking breaks
+    backTime -= noMovementTimeBack;
+    println(path.participantNumber + ", " + forwardTime + ", " + backTime + ", " + mistakes + ", " + shortcuts + ", " + wanderToAidTimes.size() + ", " + wanderTimeSum  + ", " + finishedString);
+    if(printBreakTimes)
+    {
+      println("Breaks taken: " + breakCount + " Standing still time forward: " + noMovementTimeForward + " Standing still time back: " + noMovementTimeBack);
+    }
+  }
+  
+  // find all the time based variables - forwardTime, backTime, isBackTimeSaved
+  private void findTimes()
+  {
+    String[] previousCoords = null;        //coordinates from the previous point in the path
+    float noMovementTime = 0;              //for how long the player has not moved
+    boolean isTakingBreak = false;         //is the player currently standing still     
+    float firstFrameTime = -1.0;           //when did the first frame of gameplay happen
+    float frameTime = 0;                   //time of the current frame
+    boolean isForwardTimeSaved = false;    //has forward time been saved
+    for(int i = 0; i < path.gameplayData.length; i++)
+    {
+      String coords[] = path.gameplayData[i];
+      if(coords.length > 2)    //if coordinates, not mistake and shortcut count
+      {
+        if(previousCoords != null)
+        {
+            if(previousCoords[0].equals(coords[0]) && previousCoords[1].equals(coords[1]) && previousCoords[2].equals(coords[2])     //If movement and rotational data didnt change over frame, we assume player is standing still not doing anything
+              && previousCoords[3].equals(coords[3]) && previousCoords[4].equals(coords[4]) && previousCoords[5].equals(coords[5]) && previousCoords[6].equals(coords[6]))
+            {
```

Comment:
this is way too long and bad practice. Let's have it this kind of function, which could be added instead of that if argument maybe?

private boolean hasPlayerStopped(String[] previousCoords, String[] coords){
    for(int i=0; i<coords.length; i++)
    {
      if (!previousCoords[i].equals(coords[i])){
        return false;
      }
    }
    return true;
  }

Reply:
and then you'd have this + the previous private function
```suggestion
            if(hasPlayerStopped(previousCoords, coords))
            {
```